### PR TITLE
Update Clang version for RHEL

### LINKED ISSUE #2982: RHEL Ansible Failing CI/CD
RHEL8's ansible scripts are failing to set up currently, This is suspected due to a compiler change in the default clang that ships with RHEL8,

This is impacting the rebuildable infrastructure of the CI/CD as can be seen [here](https://oe-jenkins-tf.westeurope.cloudapp.azure.com/blue/organizations/jenkins/CI-CD_Infrastructure%2FOpenEnclave-Build-Azure-Managed-Images/detail/OpenEnclave-Build-Azure-Managed-Images/187/pipeline)

### PR DESCRIPTION
**Pull request by rs--**
Fixes #2982 

Added clang version to RHEL build.

Signed-off-by: Rob Sanchez <rosan@microsoft.com>

### CODE REVIEW

File: scripts/ansible/roles/linux/openenclave/vars/redhat/main.yml [hunk 1]
```diff
@@ -30,7 +30,7 @@ yum_packages:
   - "libcurl-devel"
   - "libX11-devel"
   - "ncurses-devel"
-  - "clang"
+  - "clang-8.0.1"
```
Comment (jhand2) [inline scripts/ansible/roles/linux/openenclave/vars/redhat/main.yml:33]: What version do you get if you don't specify this?

Comment (BRMcLaren) [reply scripts/ansible/roles/linux/openenclave/vars/redhat/main.yml:33]: 9

**Pull request merged**